src/notebook/manager.ts

```diff
@@ -297,6 +309,99 @@ class NotebookManager {
   }
 
   /**
+   * Select the next cell.
+   */
+  selectNext(): void {
+    if (this.model.activeCellIndex === this.model.cells.length - 1) {
+      return;
+    }
+    this.model.activeCellIndex += 1;
+    this.deselectCells();
+  }
+
+  /**
+   * Select the previous cell.
+   */
+  selectPrev(): void {
+    if (this.model.activeCellIndex === 0) {
+      return;
+    }
+    this.model.activeCellIndex -= 1;
+    this.deselectCells();
+  }
+
+  /**
+   * Extend the selection to the previous cell.
+   */
+  extendSelectionAbove(): void {
+    let model = this.model;
+    let cells = model.cells;
+    // Do not wrap around.
+    if (model.activeCellIndex === 0) {
+      return;
+    }
+    let current = cells.get(model.activeCellIndex);
+    model.select(current);
```

Comment:
It looks like you can hold off selecting until the else statement below, because the first thing you do if the previous cell is selected is to unselect the current cell.